src/Microsoft.OpenApi/Writers/OpenApiYamlWriter.cs

```diff
@@ -241,7 +243,18 @@ public override void WriteJsonSchema(JsonSchema schema)
                                 .Build();
 
             var yamlSchema = serializer.Serialize(yamlNode);
-            WriteRaw(yamlSchema);
+
+            //remove trailing newlines
+            yamlSchema = yamlSchema.Trim();
```

Comment:
same comment about using a potential serializer option and delegating the concern to the serializer instead of mixing concerns